Weather/Weather/Controllers/PageViewController.swift

```diff
@@ -0,0 +1,163 @@
+        setViewControllers([controller], direction: .forward, animated: true, completion: nil)
+    }
+    
+    @objc func tapsOnAdd() {
+        currentController?.add()
+    }
+    
+    @objc func tapsOnDelete() {
+        currentController?.delete()
+    }
+    
+    func updateData(type: TypeOperation) {
+        switch type {
+        case .append:
+            guard let city = self.citiesModel.cities.last else { return }
+            self.models.append(WeatherModelFactory.getModel(type: city))
+            guard let controller = createController(index: models.count-1) else { return }
+            toolBar.isHidden = false
+            setViewControllers([controller], direction: .forward, animated: true, completion: nil)
+            
+        case .delete(let index):
+            self.models.remove(at: index)
+            if let controller = createController(index: 0) {
+                setViewControllers([controller], direction: .forward, animated: true, completion: nil)
+            } else {
+                guard let controller = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: NewCityController.reuseIdentifier) as? NewCityController else { return }
+                controller.modelDelegate = citiesModel
+                toolBar.isHidden = true
+                setViewControllers([controller], direction: .forward, animated: true, completion: nil)
+            }
+        }
+    }
+    
+    private func createController(index: Int) -> UIViewController? {
+        guard let controller = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: WeatherCityController.reuseIdentifier) as? WeatherCityController else { return nil }
+        controller.modelDelegate = citiesModel
+        
+        if !models.indices.contains(0) {
+            guard let controller = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: NewCityController.reuseIdentifier) as? NewCityController else { return nil }
+            controller.modelDelegate = citiesModel
+            toolBar.isHidden = true
+            return controller
+        } else if models[index] != nil {
+            controller.model = models[index]
+        } else {
+            models[index] = WeatherModelFactory.getModel(type: citiesModel.cities[index])
+            controller.model = models[index]
+        }
+        return controller
+    }
+    
+    private func getIndex(city: WeatherModelProtocol) -> Int? {
+        for i in 0..<citiesModel.cities.count {
+            switch citiesModel.cities[i] {
+            case .city(let name):
+                if name == city.city {
+                    return i
+                }
+            case .location:
+                if city as? WeatherLocationModel != nil {
+                    return i
+                }
+            }
+        }
+        return nil
+    }
+    
+    private func createMenu() {
+        var items = [UIBarButtonItem]()
```

Comment:
It's better to have dynamic items depending on screen state

Reply:
Also I would suggest not to use UIToolbar, but navigation bar of UINavigationController.
It allows you not to care about toolbar layout